src/backend/distributed/utils/directory.c

```diff
@@ -0,0 +1,215 @@
+/*-------------------------------------------------------------------------
+ *
+ * directory.c
+ *
+ * Utility functions for dealing with directories.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "postgres.h"
+#include "funcapi.h"
+#include "miscadmin.h"
+
+#include "distributed/utils/directory.h"
+
+
+/* Local functions forward declarations */
+static bool FileIsLink(const char *filename, struct stat filestat);
+
+
+/*
+ * CacheDirectoryElement takes in a filename, and checks if this name lives in
+ * the directory path that is used for job, task, table etc. files.
+ */
+bool
+CacheDirectoryElement(const char *filename)
+{
+	bool directoryElement = false;
+
+	StringInfo directoryPath = makeStringInfo();
+	appendStringInfo(directoryPath, "base/%s/", PG_JOB_CACHE_DIR);
+
+	char *directoryPathFound = strstr(filename, directoryPath->data);
+
+	/*
+	 * If directoryPath occurs at the beginning of the filename, then the
+	 * pointers should now be equal.
+	 */
+	if (directoryPathFound == filename)
+	{
+		directoryElement = true;
+	}
+
+	pfree(directoryPath);
+
+	return directoryElement;
+}
+
+
+/*
+ * CitusCreateDirectory creates a new directory with the given directory name.
+ */
+void
+CitusCreateDirectory(StringInfo directoryName)
+{
+	int makeOK = mkdir(directoryName->data, S_IRWXU);
+	if (makeOK != 0)
+	{
+		ereport(ERROR, (errcode_for_file_access(),
+						errmsg("could not create directory \"%s\": %m",
+							   directoryName->data)));
+	}
+}
+
+
+/*
+ * FileIsLink checks whether a file is a symbolic link.
+ */
+#ifdef WIN32
```

Comment:
I guess we cannot configure on windows, but maybe we remove WIN32 defs separately

Reply:
removed